mesh/mesh-db.h

```diff
@@ -93,6 +99,10 @@ struct mesh_db_prov {
 	uint8_t priv_key[32];
 };
 
+struct mesh_db_import_node {
+	parsed_type *parsed_node;
```

Comment:
That's a big :-1:. The whole idea smells - you should rather create two separate `mesh_db_import_data` structs, one for simplified flavour and the other for full import, than pass this between `mesh`, `node` and `mesh-db` modules.

Perhaps it would be even possible to reuse `mesh_db_node` for that purpose ("inherit" from it, maybe?).

Reply:
This way you wouldn't need to expose that much details in `mesh-db.h`